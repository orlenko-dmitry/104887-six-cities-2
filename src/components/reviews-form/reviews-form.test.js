import React from 'react';
import {shallow} from 'enzyme';

import ReviewsForm from '../reviews-form/reviews-form.jsx';

it(`ReviewsForm renders correctly`, () => {
  const tree = shallow(
      <ReviewsForm
        rating={0}
        comment=""
        postMessageStatus=""
        onRatingChange={() => {}}
        onCommentChange={() => {}}
        onSubmitForm={() => {}}
      />
  );

  expect(tree).toMatchSnapshot();
});
