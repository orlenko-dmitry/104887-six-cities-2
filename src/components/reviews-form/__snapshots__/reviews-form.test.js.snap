// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ReviewsForm renders correctly 1`] = `
<form
  className="reviews__form form"
  onSubmit={[Function]}
>
  <label
    className="reviews__label form__label"
    htmlFor="review"
  >
    Your review
  </label>
  <div
    className="reviews__rating-form form__rating"
  >
    <input
      className="form__rating-input visually-hidden"
      defaultValue={5}
      id="5-stars"
      name="rating"
      type="radio"
    />
    <label
      className="reviews__rating-label form__rating-label"
      htmlFor="5-stars"
      onClick={[Function]}
      title="perfect"
    >
      <svg
        className="form__star-image"
        height={33}
        width={37}
      >
        <use
          xlinkHref="#icon-star"
        />
      </svg>
    </label>
    <input
      className="form__rating-input visually-hidden"
      defaultValue={4}
      id="4-stars"
      name="rating"
      type="radio"
    />
    <label
      className="reviews__rating-label form__rating-label"
      htmlFor="4-stars"
      onClick={[Function]}
      title="perfect"
    >
      <svg
        className="form__star-image"
        height={33}
        width={37}
      >
        <use
          xlinkHref="#icon-star"
        />
      </svg>
    </label>
    <input
      className="form__rating-input visually-hidden"
      defaultValue={3}
      id="3-stars"
      name="rating"
      type="radio"
    />
    <label
      className="reviews__rating-label form__rating-label"
      htmlFor="3-stars"
      onClick={[Function]}
      title="perfect"
    >
      <svg
        className="form__star-image"
        height={33}
        width={37}
      >
        <use
          xlinkHref="#icon-star"
        />
      </svg>
    </label>
    <input
      className="form__rating-input visually-hidden"
      defaultValue={2}
      id="2-stars"
      name="rating"
      type="radio"
    />
    <label
      className="reviews__rating-label form__rating-label"
      htmlFor="2-stars"
      onClick={[Function]}
      title="perfect"
    >
      <svg
        className="form__star-image"
        height={33}
        width={37}
      >
        <use
          xlinkHref="#icon-star"
        />
      </svg>
    </label>
    <input
      className="form__rating-input visually-hidden"
      defaultValue={1}
      id="1-stars"
      name="rating"
      type="radio"
    />
    <label
      className="reviews__rating-label form__rating-label"
      htmlFor="1-stars"
      onClick={[Function]}
      title="perfect"
    >
      <svg
        className="form__star-image"
        height={33}
        width={37}
      >
        <use
          xlinkHref="#icon-star"
        />
      </svg>
    </label>
  </div>
  <textarea
    className="reviews__textarea form__textarea"
    disabled={false}
    id="review"
    name="review"
    onChange={[Function]}
    placeholder="Tell how was your stay, what you like and what can be improved"
    value=""
  />
  <div
    className="reviews__button-wrapper"
  >
    <p
      className="reviews__help"
    >
      To submit review please make sure to set 
      <span
        className="reviews__star"
      >
        rating
      </span>
       and describe your stay with at least 
      <b
        className="reviews__text-amount"
      >
        50 characters
      </b>
      .
    </p>
    <button
      className="reviews__submit form__submit button"
      disabled={true}
      type="submit"
    >
      Submit
    </button>
  </div>
</form>
`;
